### PR TITLE
feat: add disabledFeatures to widget config

### PR DESCRIPTION
**Pull request by samobasquiat**

### CODE REVIEW

**Comment by Ikari-Shinji-re** [comment]
If we disable a feature, the persisted values should not affect the application, and default values should be used instead . (like: disabledLiquiditySources, theme and language)

**Comment by Ikari-Shinji-re** [comment]
I suggest moving the settings state to the app store as a slice and initializing it with widget config, similar to the data slice. Additionally, if the feature is disabled, reset the persisted field. Also, consider setting ‘auto’ mode as the default value for the theme if it is disabled.

File: widget/embedded/src/types/config.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ export type WidgetColors = {
...
 };
 
+export enum DisabledFeatures {
+  Notifications = 'notifications',
+  Theme = 'theme',
+  Liquidity_Sources = 'liquiditySources',
+  Connect_Wallet = 'connectWallet',
+  Language = 'language',
+}
+
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/types/config.ts:31]: it is a common practice to use all caps for TypeScript enums.

File: widget/embedded/src/components/HeaderButtons/HomeButtons.tsx [hunk 2]
```diff
@@ -26,6 +28,12 @@ export function HomeButtons(props: HomeButtonsPropTypes) {
     onClickNotifications,
   } = props;
 
+  const { config } = useAppStore();
+
+  const disabledNotifications = !!config?.disabledFeatures?.includes(
+    DisabledFeatures.Notifications
+  );
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/HeaderButtons/HomeButtons.tsx:35]: A helper function that verifies disabled features could be useful.
(please check other files)

File: widget/embedded/src/types/config.ts (hunk 1 above)
Comment (samobasquiat) [reply widget/embedded/src/types/config.ts:31]: fixed.

File: widget/embedded/src/components/HeaderButtons/HomeButtons.tsx (hunk 2 above)
Comment (samobasquiat) [reply widget/embedded/src/components/HeaderButtons/HomeButtons.tsx]: fixed.

**Review by Ikari-Shinji-re — APPROVED** [review]

**Pull request merged**